arangod/Graph/Cache/RefactoredTraverserCache.cpp

```diff
@@ -110,12 +110,22 @@ bool RefactoredTraverserCache::appendEdge(EdgeDocumentToken const& idToken,
       col->getPhysical()
           ->read(_trx, idToken.localDocumentId(),
                  [&](LocalDocumentId const&, VPackSlice edge) -> bool {
-                   // NOTE: Do not count this as Primary Index Scan, we counted
-                   // it in the edge Index before copying...
-                   if constexpr (std::is_same_v<ResultType, aql::AqlValue>) {
-                     result = aql::AqlValue(edge);
-                   } else if constexpr (std::is_same_v<ResultType, velocypack::Builder>) {
-                     result.add(edge);
+                   if (onlyId) {
```

Comment:
Performance Note:
This can be applied as a template parameter to avoid some runtime-checks.
Cann't say yet if it is with it.